function-images/tests/producer-consumer/producer/producer.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+
+	"google.golang.org/grpc"
+
+	pb "github.com/MBaczun/producer-consumer/prodcon"
+)
+
+type producerServer struct {
+	producerClient pb.Producer_ConsumerClient
+	pb.UnimplementedClient_ProducerServer
+}
+
+func (ps *producerServer) ProduceStrings(c context.Context, count *pb.Int) (*pb.Empty, error) {
+	if count.Value <= 0 {
+		return new(pb.Empty), nil
+	} else if count.Value == 1 {
+		produceSingleString(ps.producerClient, fmt.Sprint(rand.Intn(1000)))
+	} else {
+		wordList := make([]string, int(count.Value))
+		for i := 0; i < int(count.Value); i++ {
+			wordList[i] = fmt.Sprint(rand.Intn(1000))
+		}
+		produceStreamStrings(ps.producerClient, wordList)
+	}
+	return new(pb.Empty), nil
+}
+
+func produceSingleString(client pb.Producer_ConsumerClient, s string) {
+	ack, err := client.ConsumeSingleString(context.Background(), &pb.String{Value: s})
+	if err != nil {
+		log.Fatalf("client error in string consumption: %s", err)
+	}
+	fmt.Printf("(single) Ack: %v\n", ack.Value)
```

Comment:
log to a file instead of the terminal. you won't be able to see prints in knative